subsys/bluetooth/controller/ll_sw/isoal.h

```diff
@@ -362,8 +340,8 @@ struct isoal_source_session {
 	uint8_t                    burst_number;
 	uint8_t                    pdus_per_sdu;
 	uint8_t                    max_pdu_size;
-	uint32_t                   latency_unframed;
-	uint32_t                   latency_framed;
+	int32_t                    latency_unframed;
+	int32_t                    latency_framed;
```

Comment:
Can the latency be negative?

Reply:
@niag-Oticon?

Reply:
In the case of the Central role (BT_CONN_ROLE_CENTRAL @ line 210)  looks to be that it can be. But I would say that this represents the best interpretation of the specs we have at present. Dissenting views and alternatives are always welcome.

Reply:
@niag-Oticon Line 210 in which file? 

Can you refer to the spec where it may indicate that that the latency can be negative?

Reply:
Sorry for the omission. It's isoal.c line 211. But to get the context of it you'll have to look at the comments from line 177 onward. Broadly the comments there are collated specs from the BT Core Specification V5.3 : Part G ISO-AL : 3. Time_Stamp and Time_Offset and they are distributes throughout the section.

However if we take one point, for example P --> C using unframed PDUs:

> SDU_Synchronization_Reference = CIS reference anchor point + CIS_Sync_Delay - CIG_Sync_Delay - 
((ISO_Interval ÷ SDU_Interval)-1) × SDU_Interval

For implementation this has been split in to CIS reference anchor point + Latency [CIS_Sync_Delay - CIG_Sync_Delay - 
((ISO_Interval ÷ SDU_Interval)-1) × SDU_Interval]. The "Latency" here will be negative. The reason for the split as I see it is that the "Latency" component can be pre-calculated as a constant, whereas the anchor point is based on the corrected PDU time stamp.

Reply:
I see. With the equation
```
SDU_Synchronization_Reference = CIS reference anchor point + CIS_Sync_Delay - CIG_Sync_Delay - ((ISO_Interval ÷ SDU_Interval)-1) × SDU_Interval
```

where you've defined 
```
latency = CIS_Sync_Delay - CIG_Sync_Delay - ((ISO_Interval ÷ SDU_Interval)-1) × SDU_Interval
```
and then
```
SDU_Synchronization_Reference = CIS reference anchor point + latency
```
then it makes sense to have `latency` as a signed value, if and only if 
```
CIS_Sync_Delay < CIG_Sync_Delay + ((ISO_Interval ÷ SDU_Interval)-1) × SDU_Interval
```
Which I do not know will ever be the case, as that also implies that
```
SDU_Synchronization_Reference > CIS reference anchor point
```
but at this point I will have to take your word for it. 

One other question: What is the relation between this `latency` and the `transport_latency`, but the latter can surely not be negative as per the HCI spec?

Reply:
First off it's not something that I implemented, but I don't see anything wrong with the implementation either.

With regard to:
`CIS_Sync_Delay < CIG_Sync_Delay + ((ISO_Interval ÷ SDU_Interval)-1) × SDU_Interval`
I think this is a given because CIS_Sync_Delay has to be less that even CIG_Sync_Delay on its own by definition of the concept.

With regard to:
`SDU_Synchronization_Reference > CIS reference anchor point`
I don't think that was quite right. Because, `SDU_Synchronization_Reference = CIS reference anchor point + Latency` where latency in this case is negative so  `SDU_Synchronization_Reference < CIS reference anchor point`. I think this is probably correct if you look at Figure 3.2 on page 3051 of the BT Core specs for the P --> C direction.

With regard to transport latency yes that must be positive. However aside from the formulae that calculate it and its presence in the diagram, I didn't see any direct application for the transport latency in reconstructing the time stamps. But I haven't played around with the formulae to see if the reconstruction of the SDU Sync Reference has incorporated it.

Reply:
Regarding transport latency, I just wanted to make sure that this (signed) `latency` field is not the same as the transport latency :) 

I don't see anything wrong, except that negative latency, to me, simply sounds odd :D

Reply:
I guess the problem was in calling it latency. We'll look into a better name, but I think it is preexisting code and not introduced in this PR, so we would like to revise the name in a later PR.